src/freenet/node/CHKInsertHandler.java

```diff
@@ -197,7 +197,7 @@ private void realRun() {
             synchronized(sender) {
                 try {
                 	if(sender.getStatus() == CHKInsertSender.NOT_FINISHED)
-                		sender.wait(5000);
+                		sender.wait(500);
```

Comment:
CHKInsertSender notifies all waiter when the status (or receivedRejectedOverload) changes. From what I can tell reducing this timeout value only increases the frequency of the `while (true)` loop this is in, but should have no other measurable effect.